modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/input_localized.js

```diff
@@ -230,6 +230,20 @@ AUI.add(
 					return '#' + namespace + id + '_' + languageId;
 				},
 
+				_moveDefaultLanguageFlagAtFirstPosition(defaultLanguageId) {
+					var instance = this;
+
+					var flags = instance._flags.getDOMNode();
+
+					var languageNode = flags.querySelector(
+						'[data-languageid="' + defaultLanguageId + '"]'
+					).parentElement;
```

Comment:
To ease our (eventual) transition to TypeScript, you should always check the result you get back from methods like `querySelector()` and friends. In this case, [its type signature](https://github.com/microsoft/TypeScript/blob/fd0ad2985b6d88e190dc757abac5a37ba102abb9/src/lib/dom.generated.d.ts#L11321-L11323) indicates that it may return a `null`-ish value, so you would add an `if (languageNode)` guard in order to keep the type checker happy.

Reply:
Er, and obviously, you can't access `.parentElement` on something that may be null, so that would be `?.parentElement`.

Reply:
@wincent do we transpile these files? I didn't want to include fancy syntax just in case

Reply:
We transpile pretty much everything at this point. There are rare exceptions, but you will struggle to find them. For example:

- Theme JS (goes through separate build process).
- Rare files that get ignored in `.npmbundlerrc` files ([example](https://github.com/liferay/liferay-portal/blob/ecb3c6255b7625cad9b94acff082ae101ec34742/modules/apps/frontend-js/frontend-js-web/.npmbundlerrc#L5-L16)) — these are generally things like AUI config files [like this one](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/loader/config.js) that get broken if Babel messes with them and we wrap them in AMD modules.

I'd guess we transpile > 98% of JS in liferay-portal at this point.